src/region.c

```diff
@@ -191,11 +191,17 @@ gdip_is_region_empty (const GpRegion *region, BOOL allowNegative)
 	case RegionTypeInfinite:
 		return FALSE;
 	case RegionTypePath:
-		/* check for an existing, but empty, path list */
 		if (!region->tree)
 			return TRUE;
-		if (region->tree->path)
-			return (region->tree->path->count == 0);
+		if (region->tree->path) {
+			if (region->tree->path->count == 0)
+				return TRUE;
+			
+			// Open paths are empty.
+			if ((region->tree->path->types[region->tree->path->count - 1] & PathPointTypeCloseSubpath) == 0)
```

Comment:
nit: Would be nice to use `if (!gdip_path_closed (region->tree->path))` here.

Reply:
Fixed :) /cc @marek-safar

Reply:
Fixed :) /cc @marek-safar